### PR TITLE
Move impala op reg batch 4

### PR DESCRIPTION
**Pull request by matthewmturner**
@datapythonista 

Last batch.

### CODE REVIEW

File: ibis/backends/bigquery/compiler.py
Comment (datapythonista) [inline ibis/backends/bigquery/compiler.py:24]: Surely not in this PR, but those will eventually need to be moved and renamed. We'll want a `BaseSelect` in `base_sql` to be imported here, in spark... and impala will also need to import and subclass them.

**Review by datapythonista — APPROVED** [review]
Thanks @matthewmturner, amazing stuff, really happy to have all this cleaned up. :)

**Comment by datapythonista** [comment]
The problems in the CI look like actual errors, if you can have a look...

**Comment by matthewmturner** [comment]
think i found the issue(needed to add the `base_sql` `binary_infix_ops` to its operation registry. hope CI is green now.

**Comment by datapythonista** [comment]
CI is red, you need to black `ibis/ibis/ibis/impala/compiler.py`.

**Comment by matthewmturner** [comment]
@datapythonista i think were good besides azp

**Comment by datapythonista** [comment]
The errors in azure seem to be actual errors: `AttributeError: module 'ibis.impala.compiler' has no attribute 'fixed_arity'`

https://dev.azure.com/ibis-project/ibis/_build/results?buildId=3732&view=logs&j=7aad0499-abbb-5e78-59a5-3e954b34383c&t=e3697a0d-8535-539a-c257-454c7fc608e9&l=462

Can you have a look please?

**Comment by matthewmturner** [comment]
@datapythonista all green!

**Comment by datapythonista** [comment]
Amazing, thanks a lot @matthewmturner, great job. I'll let @jreback review, but once this is merged you can merge master in #2435, and see if we can finally move the Impala backend without weird errors.

**Comment by matthewmturner** [comment]
sounds good, will work on that as soon as this is merged.  did you also want to get the moves of `ImpalaSelect`, `ImpalaTableSetFormatter` etc to `base_sql` in the upcoming release?

**Comment by datapythonista** [comment]
Yes, I'd like to get all decoupling, and then release.

**Comment by jreback** [comment]
thanks @matthewmturner yes very nice

**Pull request merged**
